COMETwebapp/ViewModels/Components/Shared/Selectors/ParameterSwitchKindSelectorViewModel.cs

```diff
@@ -46,21 +48,38 @@ public class ParameterSwitchKindSelectorViewModel : ReactiveObject, IParameterSw
         /// <summary>
         /// Creates a new instance of type <see cref="ParameterSwitchKindSelectorViewModel" />
         /// </summary>
-        /// <param name="switchKind">The <see cref="ParameterSwitchKind"/></param>
+        /// <param name="switchKind">The <see cref="ParameterSwitchKind" /></param>
         /// <param name="isReadOnly">The readonly state</param>
         public ParameterSwitchKindSelectorViewModel(ParameterSwitchKind switchKind, bool isReadOnly)
         {
+            this.InitialSwitchValue = switchKind;
             this.SwitchValue = switchKind;
             this.IsReadOnly = isReadOnly;
         }
 
+        /// <summary>
+        /// The initial <see cref="ParameterSwitchKind"/>
+        /// </summary>
+        public ParameterSwitchKind InitialSwitchValue { get; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="EventCallback{TValue}" /> to be called to perfom an update
+        /// </summary>
+        public EventCallback OnUpdate { get; set; }
+
         /// <summary>
         /// Get or set the <see cref="ParameterSwitchKind" />
         /// </summary>
         public ParameterSwitchKind SwitchValue
         {
             get => this.switchValue;
-            set => this.RaiseAndSetIfChanged(ref this.switchValue, value);
+            set
```

Comment:
this is a bit of an ani-pattern. I prefer we do not have extra logic in the getters/setters